### PR TITLE
Enable simple sudo for groups

### PR DESCRIPTION
**Pull request by epcim**
tested on site
enables simple sudo for groups

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -54,6 +53,67 @@ Linux with system users, sowe with password set
...
+        support-team:
+          enabled: true
+          name: support-team
+          sudo:
+            enabled: true
+            aliases:
+                RIL_SHELLS: ${_param:sudo_shells}
+                RIL_RESTRICTED: ${_param:sudo_restricted_su}
+            commands:
+                - ALL
```
Comment (fpytloun) [inline README.rst]: Make it possible to choose whether to use possibly insecure NOPASSWD, eventually other options.

File: linux/files/sudoerg [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# managed by salt
```
Comment (fpytloun) [inline linux/files/sudoerg]: Check files/hosts or resolv.conf for nicer header about managed by salt and do not edit.

File: linux/system/group.sls [hunk 3]
```diff
@@ -14,6 +14,9 @@ system_group_{{ group_name }}:
   {%- if group.gid is defined and group.gid %}
   - gid: {{ group.gid }}
   {%- endif %}
+  - require_in:
+    - file: /etc/sudoers.d/91-salt-groups
```
Comment (fpytloun) [inline linux/system/group.sls]: It would be best to have one file per group to avoid one huge file.

File: README.rst [hunk 4]
```diff
@@ -54,6 +53,70 @@ Linux with system users, sowe with password set
...
+          sudo:
+            enabled: true
+            aliases:
+                SUPPORT_SHELLS: ${_param:sudo_shells}
+                SUPPORT_RESTRICTED: ${_param:sudo_restricted_su}
+            commands:
+                - ALL
+                - '!SUPPORT_SHELLS'
+                - '!SUPPORT_RESTRICTED'
+            hosts: ALL
```
Comment (fpytloun) [inline README.rst]: Should be list or key should be `host`

File: README.rst [hunk 5]
```diff
@@ -54,6 +53,110 @@ Linux with system users, sowe with password set
...
+            PRODUCTION:
+            - db1
+            - db2
+          runas:
+            DBA:
+            - postgres
+            - mysql
+            SALT:
+            - root
+          command:
```
Comment (fpytloun) [inline README.rst]: commands are also kind-of aliases or no?

File: README.rst [hunk 6]
```diff
@@ -54,6 +53,110 @@ Linux with system users, sowe with password set
...
+            SUPPORT_SHELLS: ${_param:sudo_shells}
+            SUPPORT_RESTRICTED: ${_param:sudo_restricted_su}
+        users:
+          saltuser1: {}
```
Comment (fpytloun) [inline README.rst]: This shouldn't do anything, right?

Comment (epcim) [inline README.rst]: no, add default values = saltuser all=(all) all - > I have a check for that in other comit

File: README.rst (hunk 5 above)
Comment (epcim) [inline README.rst]: yes,, host_alias, cmd_alias, runas_alias... (I don't use plural as it's more confusing) as you have then in users/groups sec. commands. Also "runass, runasses" or what it could be.

File: tests/pillar/system.sls [hunk 7]
```diff
@@ -75,11 +107,11 @@ linux:
       opencontrail:
         source: "deb http://ppa.launchpad.net/tcpcloud/contrail-2.20/ubuntu trusty main"
         architectures: amd64
-    policyrcd:
-      - package: cassandra
-        action: exit 101
-      - package: '*'
-        action: switch
+    #policyrcd:
```
Comment (fpytloun) [inline tests/pillar/system.sls]: You should probably uncomment back blocks of testing pillar before merging.

Comment (epcim) [inline tests/pillar/system.sls]: kitchen will not pass anyway - tested locally, not part of this commit., it this case salt complains for missing priority to be set. Policyrcd had a different issues I challenged. I want to have a look on this on the kitchen commit.

File: README.rst [hunk 8]
```diff
@@ -31,7 +30,7 @@ Basic Linux box
         timezone: 'Europe/Prague'
         utc: true
 
-Linux with system users, sowe with password set
+Linux with system users, some with password set
```
Comment (cznewt) [inline README.rst]: please use just pillars here

File: tests/pillar/system.sls [hunk 9]
```diff
@@ -2,10 +2,11 @@ linux:
   system:
     enabled: true
     cluster: default
-    name: test01
+    name: linux
```
Comment (cznewt) [inline tests/pillar/system.sls]: don't append this file add system_sudo.sls instead

**Comment by pjediny** [comment]
What about `visudo -cf` to check if the configuration is correct. 
https://blog.afoolishmanifesto.com/posts/checking-sudoers-with-visudo-in-saltstack/

File: tests/pillar/system.sls (hunk 9 above)
Comment (epcim) [inline tests/pillar/system.sls]: we don't want to have 30 unit tests to spin 30 containers with CI. So we tend to have test/pillar for single/cluster/system/storage/network and possibly no more. I did this changes as I locally already test the formula with kitchen.

**Comment by epcim** [comment]
[**@pjediny**](https://github.com/pjediny) commented on [Mar 3, 2017, 12:16 PM GMT+1](https://github.com/salt-formulas/salt-formula-linux/pull/64#issuecomment-283929275 "2017-03-03T11:16:26Z - Replied by Github Reply Comments"):
> What about `visudo -cf` to check if the configuration is correct. 
> https://blog.afoolishmanifesto.com/posts/checking-sudoers-with-visudo-in-saltstack/

thanks, will add it: btw it will broke part of the sudo if it's wrong anyway, as it probably render the file in under /etc/sudoers.d/ not in temp. directory..
